app/utils/minimal-include.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Returns the a comma-separated string containing the minimal necessary include
+ * paths. When a path is a complete prefix of another path, the shorter path
+ * doesn't need to be part of the include parameter, so it gets removed by this
+ * function.
+ *
+ * The function works by iterating over each path and storing only paths that
+ * aren't a prefix of other paths. For each path, we iterate over all other
+ * paths and make sure that (1) the current path is not a prefix of another path
+ * and (2) that other paths are not full prefix of the current path. If (1), we
+ * discard the current path, if (2) we discord the other path.
```

Comment:
Can we maybe simplify by first sorting the `paths` array on string length? As such we only need to check (1) as (2) cannot occur then.